### DISCUSSION TITLE
Add my own honeypot to TPot

### DISCUSSION THREAD

**Discussion by maurodelossantos** [telekom-security/tpotce#1447] [Q&A]
Hello! I'd like to **add my own honeypot to TPot** to expose some vulnerable services that will require some extra manual exploitation (BOFs, customized vulnerabilities, etc.) to later compare the results between automated honeypots and customized ones. This question is to ask/confirm the **steps to follow** to add my honeypot[s] to TPot and be able to visualize in the _WebGUI_ its results:

1. Add a folder called `myHoneypot/` under the `docker/` with a `dist/` subfolder for configuration files if needed, a `Dockerfile`, and a `docker-compose.yml` to spin up the container with the vulnerable[s] service[s].
2. Adjust the `tpot.yml` and the main `docker-compose.yml` (?) I understand now this isn't needed as the file is going to be updated dynamically with the content of `Dockerfile `and `docker-compose.yml` under the `myHoneypot/` folder. Still want to corroborate this.
3. Edit the `docker/elk/logstash/dist/logstash.conf` file to add the logs of our honeypot. I copied the changes made for Log4Pot in [this commit](https://github.com/telekom-security/tpotce/commit/b0339610a21f9a0c1203dcfdd23a4d201122c381#diff-652c3c66426f704379dbe93b7fae08d9f09127fc61f4276e8f5c10a90b80a279). Is this correct?
4. Modify the Kibana objects. Completely lost about this step. I've seen [this commit](https://github.com/telekom-security/tpotce/commit/ceee197e6881588645d33161ed2af044d4d5fbc7) modifying them for adding Log4Pot but as they are binary files I can't really see what to change.
5. Based on some commits (especially the ones to add `Log4Pot `-- [_fb93d851190109aadddda2c834c81107e2559583_](https://github.com/telekom-security/tpotce/commit/fb93d851190109aadddda2c834c81107e2559583)), I've tried to debug which extra files I should edit. Are any of those worthy/required to modify?:  `/etc/compose/log4j.yml` / `etc/compose/standard.yml` / `logrotate/logrotate.conf` / anything else?
6. Stop and start the TPot service. (?) Maybe to apply the changes a restart isn't enough and I need to manually apply and run the container for the honeypot.

Am I missing anything? With just those steps, will I be able to **successfully deploy** and see the information about my Honeypot in the WebGUI (mainly in the attack map and Elasticsearch)?

Any help is very welcome! Thanks!

**Comment by witchermixer** [comment]
Have you been able to resolve your issue? I'm facing similar situation, but I want to add new service to already working instance of T-POT. Unfortunately addind logs to /data folder and editing logstash.conf does not help.

If anyone have any tips I would be grateful. 

Cheers.

**Comment by SimonKling** [comment]
So was a way found ?

**Comment by ASalcedoC** [comment]
Trying to revive this. Did someone find a way?

Cheers.

**Comment by pokix** [comment]
Hello! I managed to make it work!

As the author said, I created my own container in tpotce/docker/ . And I created a tpotce/data/ for the logs/payloads exports

Then I edited all those files:
-tpotce/compose/sensor.yml -> added my compose config
-tpotce/docker-compose.yml -> added my compose config
-tpotce/docker/tpotinit/dist/etc/logrotate/logrotate.conf -> added logrotate conf
-tpotce/docker/tpotinit/dist/bin/clean.sh -> init conf to add
-tpotce/docker/elk/logstash/dist/http_output.conf -> added logstash conf to read the logs
-tpotce/docker/elk/logstash/dist/logstash.conf -> added logstash conf to read the logs

Now please note that the changes made in tpotinit and logstash won't take effect until you rebuild their docker images. So here is what you need to do

-shutdown tpot : docker compose down
-list all images : docker images
-Note tpotinit and logstash names, tags and image ID
-Remove the old images using their ID : docker rmi
-Go in the folder with the Dockerfile : cd tpotce/docker/logstash (and/or tpotinit)
-Rebuild them : docker build -t : . (for example , docker build -t ghcr.io/telekom-security/logstash:24.04.1 .). And don't forget the dot in the command line!
-restart tpot : docker compose up -d
-Is that a custom log from my custom honeypot on my Kibana? My life is so beautiful!
